hail/python/hailtop/batch_client/client.py

```diff
@@ -315,7 +315,7 @@ def create_batch(self,
         return BatchBuilder.from_async_builder(builder)
 
     def update_batch(self, batch: Union[int, Batch]) -> 'BatchBuilder':
-        builder = async_to_blocking(self._async_client.update_batch(batch))
+        builder = async_to_blocking(self._async_client.update_batch(batch._async_batch))
```

Comment:
looks like this needs to be `batch._async_batch if isinstance(batch, Batch) else batch`